### PR TITLE
API Doc Created & hydrus setup.

### LINKED ISSUE #5: Create API_Doc & Setup hydrus
using the NPL JSON-LD Vocabulary create the API_Doc and Setup hydrus for POC.

### PR DESCRIPTION
**Pull request by Purvanshsingh**
Fixes #5 

I have created the API_Doc with 2 classes for now and added operations for the Loan class only.

The database is set up successfully.
Screenshot from 2021-06-08 19-58-22

creditrisk_poc

We are facing a problem that all the classes should be displayed at `http://localhost:8080/creditrisk_api/`
but we are only getting the EntryPoint.

We (@farazkhanfk7) discussed the problem, which might be due to URIs of properties & class which is not dereferenceable by hydrus.
as I have used 
`"NPL": "https://raw.githubusercontent.com/Purvanshsingh/creditrisk-poc/main/NonPerformingLoan.json"` in API_Doc context.

### CODE REVIEW

**Comment by Purvanshsingh** [comment]
POC is functionally working now ( tested with POSTMAN )

**PUT:**

Screenshot from 2021-06-10 04-16-11

**GET:**

Screenshot from 2021-06-10 04-16-46

For now, I have added operations for the Loan class only.
|
|__ **GET**
|__ **PUT**
|__ **POST**

**Comment by Mec-iS** [comment]
There are some basic Python practices that are not applied: the project structure is wrong, please follow [this structure](https://docs.python-guide.org/writing/structure/)

1. All the code has to be put in a directory called with the name of library i.e. `creditrisk_poc` under the root project; `tests` directory has to be in the root, `docs` if present has to be in the root. All code should not be in the root (only `setup.py` and other configuration or complimentary files should be in root)
2. modules names are redundant, avoid repeating `creadit_risk....` in every module name
3. `setup.py` is missing
4. module names are uppercase i.e. `Index.py` and the name is not really adeguate, should be `main.py` or similar

Please follow Python convention or otherwise the code is hardly readable, see the [official documentation packaging conventions](https://packaging.python.org/tutorials/packaging-projects/).

This is about the project strucutre. More comments on other part of the code will possible follow as review goes on.

**Review by Mec-iS — CHANGES_REQUESTED** [review]
see my comment about project structure

File: API_Doc/credit_risk_POC_docwriter.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+# creating Entrypoint
+api_doc.gen_EntryPoint()
+# generating API_Doc
+doc = api_doc.generate()
+
+# saving the API_Doc
+if __name__ == "__main__":
+    """Print the complete sample Doc in api_doc_output.py."""
+    import json
+    dump = json.dumps(doc, indent=4, sort_keys=True)
```
Comment (Mec-iS) [inline API_Doc/credit_risk_POC_docwriter.py]: What you are doing here is not the proper way.
What you want to do is:
1. build the ApiDoc programmatically (ok done)
2. save the JSON-LD string in a file (`json.dumps(doc)` -> `ApiDoc.jsonld`)
3. if you want to have the file in a Python dictionary: create a new file `apidoc.py` that does: 
```
doc = None
with open("ApiDoc.jsonld") as d:
    doc = json.loads(d.read())
```
All this has to be put under `if __name__ == "__main__":`, it tells the Python interpreter to run all the code when the command `$ python creditrisk_poc_docwriter.py` is run; as it is now only the last lines will be run but `doc` will be undefined.

Comment (Purvanshsingh) [reply API_Doc/credit_risk_POC_docwriter.py]: Okay, will fix this.
I took the reference from the sample present in the hydrus itself.
https://github.com/HTTP-APIs/hydrus/blob/develop/hydrus/samples/doc_writer_sample.py

Comment (Mec-iS) [reply API_Doc/credit_risk_POC_docwriter.py]: that's totally wrong I agree. That is from 2017 though and conditions were quite different, for example Github did not have code reviews at the time... D) (or yes I don't remember, btw every time there is code-reuse the code should be checked for working and tested)

File: API_Doc/creditrisk_api_doc.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
+"""Generated API Documentation sample using doc_writer_sample.py."""
+    
+doc = {
```
Comment (Mec-iS) [inline API_Doc/creditrisk_api_doc.py]: If you try to do: 
```
from pyld import jsonld
print(jsonld.expand(doc))
```
At some point close to the end of the print:
```
 {'http://www.w3.org/ns/hydra/core#description': [{'@value': 'The Borrower Class'}],
      'http://www.w3.org/ns/hydra/core#title': [{'@value': 'borrower'}],
      'http://www.w3.org/ns/hydra/core#property': [{'@id': 'http://localhost:8080/creditrisk_api/vocab?resource=EntryPoint/Borrower',  <<<<<
```
this link is wrong. I don't know if it is a problem of the ApiDoc or of the doc_maker

**Review by open-risk — APPROVED** [review]

**Review by Mec-iS — CHANGES_REQUESTED** [review]
As specified in the confcall, there are two main changes to apply:
* make the repository compliant with [Python Official Style guide -- PEP8](https://www.python.org/dev/peps/pep-0008/) (i.e. names for modules and packages should be lowercase), clean-up requirements.txt
* use `json` for serialisation

**Review by Mec-iS — APPROVED** [review]

**Pull request merged**